### PR TITLE
removes NewTask and NewSubtask

### LINKED ISSUE #626: Remove `NewTask` and `NewSubtask` constructor
### Describe the chore

Be explicit in `New...Aggregate`

### PR DESCRIPTION
**Pull request by PaulKalho**
## Which issues does this pull request close?
closes #626

### CODE REVIEW

File: services/tasks-svc/internal/task/aggregate/aggregate.go [hunk 1]
```diff
@@ -19,7 +19,7 @@ type TaskAggregate struct {
 }
 
 func NewTaskAggregate(id uuid.UUID) *TaskAggregate {
-	aggregate := &TaskAggregate{Task: models.NewTask()}
+	aggregate := &TaskAggregate{Task: &models.Task{}}
 	aggregate.AggregateBase = hwes.NewAggregateBase(TaskAggregateType, id)
 	aggregate.Task.ID = id
 	aggregate.initEventListeners()
```
Comment (FoseFx) [inline services/tasks-svc/internal/task/aggregate/aggregate.go:25]:
```suggestion
	aggregate := &TaskAggregate{
		AggregateBase: hwes.NewAggregateBase(TaskAggregateType, id),
		Task: &models.Task{
			ID: id,
			CreatedAt:     time.Now().UTC(),
			Status:        pb.TaskStatus_TASK_STATUS_TODO,
			AssignedUsers: make([]uuid.UUID, 0),
			Subtasks:      make(map[uuid.UUID]Subtask, 0),
		},
	}
	aggregate.initEventListeners()
```

File: services/tasks-svc/internal/task/aggregate/aggregate.go [hunk 2]
```diff
@@ -171,9 +171,11 @@ func (a *TaskAggregate) onSubtaskCreated(evt hwes.Event) error {
 		return err
 	}
 
-	subtask := models.NewSubtask()
-	subtask.ID = subtaskID
-	subtask.Name = payload.Name
+	subtask := &models.Subtask{ID: subtaskID, Name: payload.Name}
```
Comment (FoseFx) [inline services/tasks-svc/internal/task/aggregate/aggregate.go:174]:
```suggestion
	subtask := &models.Subtask{
		ID: subtaskID,
		Name: payload.Name,
		CreatedAt: time.Now().UTC(),
		Done:      false,
	}
```

File: services/tasks-svc/internal/task/aggregate/aggregate.go [hunk 3]
```diff
@@ -171,9 +171,11 @@ func (a *TaskAggregate) onSubtaskCreated(evt hwes.Event) error {
...
+
+	if a.Task.Subtasks == nil {
+		a.Task.Subtasks = make(map[uuid.UUID]models.Subtask)
+	}
```
Comment (FoseFx) [inline services/tasks-svc/internal/task/aggregate/aggregate.go:178]: we could keep this, but it's not needed anymore with my proposed change to NewAggregate (I think)

**Review by FoseFx — CHANGES_REQUESTED** [review]

File: services/tasks-svc/internal/task/aggregate/aggregate.go (hunk 3 above)
Comment (PaulKalho) [reply services/tasks-svc/internal/task/aggregate/aggregate.go]: Yup

**Review by MaxSchaefer — APPROVED** [review]

**Review by FoseFx — APPROVED** [review]

**Pull request merged**